src/MongoDB.Driver.Core/Core/Operations/UpdateOpcodeOperation.cs

```diff
@@ -216,6 +216,10 @@ private WriteConcernResult ExecuteProtocol(IChannelHandle channel, CancellationT
             {
                 throw new NotSupportedException("OP_UPDATE does not support arrayFilters.");
             }
+            if (_request.Hint != null)
+            {
+                throw new NotSupportedException("Unacknowledged OP_UPDATE does not support hints.");
+            }
```

Comment:
I am safely assuming here that OP_UPDATE is unacknowledged because the only path leading to this code is for unacknowledged write concern: [link to line](https://github.com/MikalaiMazurenka/mongo-csharp-driver/blob/712f50827c5b1b2e26d102d619e42c7b10020d22/src/MongoDB.Driver.Core/Core/Operations/UpdateOpcodeOperation.cs#L157)

Reply:
True... but the error message is still wrong.